sympy/physics/matrices.py

```diff
@@ -167,6 +171,9 @@ def mdft(n):
     ========
 
     >>> from sympy.physics.matrices import mdft
+    >>> from sympy.utilities.exceptions import SymPyDeprecationWarning
+    >>> import warnings
+    >>> warnings.simplefilter("ignore", SymPyDeprecationWarning)
     >>> mdft(3)
```

Comment:
Hm... it would be better if we could avoid changing the warnings filter. I think I have fixed it to reset between doctests but I'd still prefer not modify global settings in a doctest...

It would be good to use the `.. deprecated` sphinx directive here:
https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated

Reply:
OK! I searched the code how to handle it and that was the example I found. I'll have a look.

Reply:
I'd rather just disable the doctest than mess with the warnings filter.

Reply:
Yes, it is not obvious how to handle the doc tests for deprecated functions.

I guess it could make sense to mention the deprecation details in the documentation as well (or probably more valuable than having an example of code that shouldn't be used).

Reply:
Let's please avoid doing `warnings.simplerfilter('ignore')` in a doctest. Doctests are supposed to show users what to do. But we don't want users to ignore the warnings. 

Better options would be

- Use `# doctest: +SKIP` to skip the doctest
- Some sort of decorator we can put on the function to make the doctester ignore warnings
- Remove the documentation entirely, and replace it with a message that it is deprecated and pointing to the preferred function

And as a general rule, avoid putting code in examples that exists solely to make the doctester happy. The purpose of documentation examples is first and foremost to be informative to users. The doctester is only there to ensure the output is correct and up-to-date. Remember that doctests are not tests, rather examples that happen to be tested. I would rather have an example that isn't actually tested, but is readable and informative, than one that is testable but difficult to read and understand.

Reply:
> Some sort of decorator we can put on the function to make the doctester ignore warnings

Actually, we already are using `@deprecated` as a decorator. So we can extend it to tag the function in a way that makes the doctester handle warnings. Although I'm also happy with both other options I suggested.

Reply:
> * Remove the documentation entirely, and replace it with a message that it is deprecated and pointing to the preferred function

This is really what seems to make sense here in this case, I believe. I'll remove it, point to the correct function and improve the documentation of `DFT` based on the (actually quite good, at least in comparison) documentation of `mdft`.